### ISSUE TITLE
KeyError: 'become_pass'

### ISSUE THREAD

**Issue by nenych** [ansible/ansible#17301]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

task_executor
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

```
##### CONFIGURATION

[defaults]
log_path = /var/log/ansible.log
##### OS / ENVIRONMENT

ubuntu 16.04
##### STEPS TO REPRODUCE

1) apt install python3-pip
2) pip3 install ansible
3) copy example from official documentation (" Python API 2.0 ") into file test.py
inside example  change path to hosts file
also I added callback function v2_runner_on_failed:
    def v2_runner_on_failed(self, result, ignore_errors=False):
        if 'exception' in result._result:
            # Extract the error message and log it
            error = result._result
            print(error)

4) python3 test.py

I tried to run another examples but result was with same fail also. It works in python 2.7, but i have to use python3

```

```

##### EXPECTED RESULTS

OK Status
##### ACTUAL RESULTS

{'msg': 'Unexpected failure during module execution.', 'exception': 'Traceback (most recent call last):\n  File "/opt/ansibleAPI/ansibleEnv/lib/python3.5/site-packages/ansible/executor/task_executor.py", line 124, in run\n    res = self._execute()\n  File "/opt/ansibleAPI/ansibleEnv/lib/python3.5/site-packages/ansible/executor/task_executor.py", line 353, in _execute\n    self._play_context = self._play_context.set_task_and_variable_override(task=self._task, variables=variables, templar=templar)\n  File "/opt/ansibleAPI/ansibleEnv/lib/python3.5/site-packages/ansible/playbook/play_context.py", line 312, in set_task_and_variable_override\n    if hasattr(task, attr):\n  File "/opt/ansibleAPI/ansibleEnv/lib/python3.5/site-packages/ansible/playbook/base.py", line 115, in _generic_g\n    value = self._attributes[prop_name]\nKeyError: \'become_pass\'\n', 'stdout': '', 'failed': True}

I don't use become, why my error always: KeyError: 'become_pass'?

```

```

**Comment by ryansb** [comment]
@abadger looks like some kind of py3-specific argument handling. Have you seen this?

**Comment by nenych** [comment]
Thanks for Your response. Solution is adding become_pass=None and docker_extra_args=None to options. After that i ran into new problems:  write() argument must be str, not bytes
To resolve first write case I had to change f.write(zipdata) to f.write(str(zipdata)) in 603's row inside  <python library>/ansible/executor/module_common.py
Unfortunately it was only one of many similar cases.

**Comment by abadger** [comment]
@25ar Ansible doesn't yet run on python3.  There is work being done to get 2.2.0 to a tech preview level of functionality with python3 (ie: you'll be able to do simple things but there will definitely be minor bugs in core features and some more peripheral things flat out will be broken.)

If you want to test it out and maybe get bugs reported or fixed in time to get into 2.2 (if it misses 2.2, we'll get them into bugfixes of 2.2 or 2.3 if the change required is more invasive), you can try running the devel branch of the git repository.  Either via http://docs.ansible.com/ansible/intro_installation.html#running-from-source  or by pip installing from the repository: http://docs.ansible.com/ansible/intro_installation.html#latest-releases-via-pip

Note -- We use git submodules for ansible's modules.  I haven't used pip to install ansible from the git repo so I'm not sure if it automatically checks out the submodules or not.

**Comment by abadger** [comment]
Also note -- the Python API would be considered a peripheral feature... there may be a lot more bugs trying to use that with Python3 than trying to achieve your goals via the command line programs.

I'm going to close this bug as working on Python3 support is being worked on for Ansible-2.2 and this is reported against 2.1.  Feel free to open new bugs as you find issues with running the devel branch on python3.  Thanks!
